src/Microsoft.IdentityModel.JsonWebTokens/JsonWebTokenHandler.cs

```diff
@@ -1146,7 +1146,8 @@ public virtual TokenValidationResult ValidateToken(string token, TokenValidation
                     {
                         SecurityToken = jwtToken,
                         ClaimsIdentity = innerTokenValidationResult.ClaimsIdentity,
-                        IsValid = true
+                        IsValid = true,
+                        TokenType = innerTokenValidationResult.TokenType
```

Comment:
Not 100% sure about this one. Maybe it should be another property, like `InnerTokenType`?

Reply:
I think is should be as you mentioned the inner, as the JWE is just a wrapper without any info.